File: Python/0218. 字符串处理.py
# If you need to import additional packages or classes, please import here.

def func():
    # please define the python3 input here. 
    # For example: a,b = map(int, input().strip().split())
    # please finish the function body here.
    # please define the python3 output here. For example: print().
    while True:
        try:
            n = int(input())
            for _ in range(n):
                s = input()
                res = ['0']
                for c in s:
                    if c == res[-1]:
                        res.pop()
                    else:
                        res += c
                print("Yes" if len(res) == 1 else "No")
        except EOFError:    
            break
    
if __name__ == "__main__":
    func()

File: C++/0218. 字符串处理.cpp
// we have defined the necessary header files here for this problem.
// If additional header files are needed in your program, please import here.
int main()
{  
  // please define the C++ input here. For example: int a,b; cin>>a>>b;;  
  // please finish the function body here.  
  // please define the C++ output here. For example:cout<<____<<endl; 
    int n;
    cin>>n;
    string s;
    while(cin>>s) {
        string res="0";
        for(auto c:s) {
            if(c == res.back()) {
                res.pop_back();
            } else {
                res += c;
            }
        }        
        cout<<(res.size() == 1 ? "Yes" : "No")<<endl;
    }
    return 0;
}
